copycat/api/src/main/java/org/apache/kafka/copycat/data/CopycatSchema.java

```diff
@@ -0,0 +1,235 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.copycat.data;
+
+import org.apache.kafka.copycat.errors.DataException;
+
+import java.nio.ByteBuffer;
+import java.util.*;
+
+public class CopycatSchema implements Schema {
+    private static final Map<Type, Class<?>> SCHEMA_TYPE_CLASSES = new HashMap<>();
+    static {
+        SCHEMA_TYPE_CLASSES.put(Type.INT8, Byte.class);
+        SCHEMA_TYPE_CLASSES.put(Type.INT16, Short.class);
+        SCHEMA_TYPE_CLASSES.put(Type.INT32, Integer.class);
+        SCHEMA_TYPE_CLASSES.put(Type.INT64, Long.class);
+        SCHEMA_TYPE_CLASSES.put(Type.FLOAT32, Float.class);
+        SCHEMA_TYPE_CLASSES.put(Type.FLOAT64, Double.class);
+        SCHEMA_TYPE_CLASSES.put(Type.BOOLEAN, Boolean.class);
+        SCHEMA_TYPE_CLASSES.put(Type.STRING, String.class);
+        SCHEMA_TYPE_CLASSES.put(Type.ARRAY, List.class);
+        SCHEMA_TYPE_CLASSES.put(Type.MAP, Map.class);
+        SCHEMA_TYPE_CLASSES.put(Type.STRUCT, Struct.class);
+        // Bytes are handled as a special case
```

Comment:
Am already working on a follow up patch for turning serializer/converter into a single class and this code has changed a bit as a result. In the new version I track the list of types and we also have the inverted map with helpers to look up and cache mappings like ArrayList -> Type.ARRAY, so I think this gets clearer in the follow up.

The drawback with standardizing on ByteBuffer is that it's inconsistent with what the rest of Kafka does -- we use byte[] in our public interfaces.